devscripts/prepare_manpage.py

```diff
@@ -43,6 +43,27 @@ def filter_excluded_sections(readme):
         '', readme)
 
 
+def convert_code_blocks(readme):
+    out = ''
+    in_code_block = False
+
+    for line in readme.split('\n'):
+        if in_code_block:
+            if line.startswith('```'):
+                in_code_block = False
+                out += '\n'
+                continue
+            out += '    ' + line + '\n'
+            continue
+        if line.startswith('```'):
+            in_code_block = True
+            out += '\n'
+            continue
+        out += line + '\n'
+
+    return out
```

Comment:
what do you think about this? 
```suggestion
def _convert_code_blocks(readme):
    current_code_block = None

    for line in readme.splitlines(True):
        if current_code_block:
            if line == current_code_block:
                current_code_block = None
                yield '\n'
            else:
                yield f'    {line}'
        elif line.startswith('```'):
            current_code_block = '`' * line.count('`')
            yield '\n'
        else:
            yield line


def convert_code_blocks(readme):
    return ''.join(_convert_code_blocks(readme))
```

Reply:
> Have you tested these changes? They don't seem to work on my end.

there was a bug. I've edited the suggestion, have tested it, and can confirm it works now.

> Also, what's up with having two functions, one calling the other? 

`_convert_code_blocks` is the generator and `convert_code_blocks` returns a string from what the generator yields